sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/BatchWrite.java

```diff
@@ -104,4 +106,14 @@ default void onDataWriterCommit(WriterCommitMessage message) {}
    * clean up the data left by data writers.
    */
   void abort(WriterCommitMessage[] messages);
+
+  /**
+   * Similar to {@link #commit(WriterCommitMessage[])}, but providing operation metrics to
+   * this batch write.
+   * @param metrics operation metrics.  The keys will be prefixed by operation type, eg `merge`
```

Comment:
I think it's better to list all the metric keys and their meanings in the API doc, as it's part of the API.

Reply:
added the javadocs